### PR TITLE
Collapse advanced account settings by default

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://github.com/sourcegraph/sourcegraph/issues/56572

Previously, the dialog to create a new account with access token showed

* (required) endpoint URL
* (required) access token
* (optional) custom HTTP headers

This PR updates the dialog to collapse advanced and optional settings like "custom HTTP headers" to make it easier for users to create an account with only an endpoint URL and access token.

## Test plan
- `./gradlew :runIde`
- Create new account
- Confirm that custom HTTP headers are collapsed

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/config/CodyLoginPanel.kt [hunk 1]
```diff
@@ -25,13 +25,10 @@ class CodyLoginPanel(
 ) : Wrapper() {
 
   private val serverTextField = ExtendableTextField(SourcegraphServerPath.DEFAULT_HOST, 0)
-  private val customRequestHeadersField = ExtendableTextField("", 0)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/config/CodyLoginPanel.kt:28]: It looks like `ExtendableTextField` must be created inside the panel to get it working properly within `collapsibleGroup`. I do not know the specific reasons - the problem seems to be related to deep internals.

That's why I needed to move the construction and expose it as a variable to `CodyLoginPanel`.

https://github.com/sourcegraph/sourcegraph/assets/19799111/87289fad-2aab-402a-b731-ecd71f98e568

**Review by Sa1to — APPROVED** [review]
LGTM

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 

@mkondratek I took the liberty to update the PR title and description to match my preferred format:

- Title: state the user-visible change (example: fix windows autocomplete, ...). The title should not explain an implementation detail.
- Description: describe the problem before this PR, and describe the state after this PR. Sometimes it can feel redundant to repeat the state after the PR but it's often a good opportunity to elaborate on the solution you went with. 

I will merge once I have figured out how to get the CI green in the other PR.

**Comment by olafurpg** [comment]
@mkondratek can you please rebase on top of main to get the CI running?

**Comment by mkondratek** [comment]
> @mkondratek can you please rebase on top of main to get the CI running?

@olafurpg, done!

**Pull request merged**